runtime/taskcontext.go

```diff
@@ -195,6 +212,15 @@ func (c *TaskContext) NewLogReader() (io.ReadCloser, error) {
 	return c.logStream.NextReader()
 }
 
+func (c *TaskContext) ExtractLog() (ioext.ReadSeekCloser, error) {
+	file, err := os.Open(c.logLocation)
```

Comment:
Hmm.... I guess it's totally overkill to suggest that we use the existing
interface... Can we at-least agree to return an ErrLogNotClosed error until
the log is closed... So we avoid races here... Obviously using this method
outside of Finished handler likely won't work...

Den 6. sep. 2016 12.26 PM skrev "Greg Arndt" <notifications@github.com>:

> In runtime/taskcontext.go
> <https://github.com/taskcluster/taskcluster-worker/pull/80#discussion_r77701783>
> :
>
> > @@ -195,6 +212,15 @@ func (c *TaskContext) NewLogReader() (io.ReadCloser, error) {
> >  	return c.logStream.NextReader()
> >  }
> >
> > +func (c *TaskContext) ExtractLog() (ioext.ReadSeekCloser, error) {
> > +	file, err := os.Open(c.logLocation)
>
> It's necessary for retrying artifact uploading if I recall correctly. It's
> been awhile since I worked on this.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/taskcluster/taskcluster-worker/pull/80/files/bc3b3c3da0f90e36bb10a488696634d6b91c4737#r77701783>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAJI5FQtyLVgU_7kkPBISa2yt_MHQEaWks5qnb5hgaJpZM4I854t>
> .
>